File: ProblemSet/Week 3/Problem/JavaScript/LinkedList/src/block.js
// block.js
const { SHA256 } = require('crypto-js');

class Block {
    constructor(timestamp, data, previousHash) {
        this.timestamp = timestamp;
        this.data = data;
        this.previousHash = previousHash;
        this.hash = this.calculateHash();
    }

    calculateHash() {
        const blockString = `${this.timestamp}${JSON.stringify(this.data)}${this.previousHash}`;
        return SHA256(blockString).toString();
    }
}

module.exports = Block;

File: ProblemSet/Week 3/Problem/Python/LinkedList-v2-OOP/blockchain_auction_system/block.py
import hashlib


class Block:
    """
    Represents a block in the blockchain.
    """
    def __init__(self, timestamp, data, previous_hash):
        self.timestamp = timestamp
        self.data = data
        self.previous_hash = previous_hash
        self.hash = self.calculate_hash()

    def calculate_hash(self):
        """
        Calculate the hash of the block.
        """
        block_string = f"{self.timestamp}{self.data}{self.previous_hash}"
        return hashlib.sha256(block_string.encode()).hexdigest()